waku/v2/protocol/waku_filter/waku_filter.nim

```diff
@@ -187,44 +188,63 @@ method init*(wf: WakuFilter) =
 
   wf.handler = handle
   wf.codec = WakuFilterCodec
+  wf.timeout = filterTimeout
 
-proc init*(T: type WakuFilter, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: MessagePushHandler): T =
+proc init*(T: type WakuFilter, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: MessagePushHandler,timeout: float = WakuFilterTimeout): T =
   let rng = crypto.newRng()
   var wf = WakuFilter(rng: rng,
                       peerManager: peerManager, 
                       pushHandler: handler)
-  wf.init()
-  
+  wf.init(filterTimeout = timeout)
   return wf
 
 proc setPeer*(wf: WakuFilter, peer: RemotePeerInfo) =
   wf.peerManager.addPeer(peer, WakuFilterCodec)
   waku_filter_peers.inc()
 
+# If we have already failed to send message to this peer,
+# check for elapsed time and if it's been too long, remove the peer.
+proc handleClientError*(wf: WakuFilter, subscriber: Subscriber){.raises: [Defect, KeyError], async.} = 
+  var subKey: string = $(subscriber)
+  if wf.failedPeers.hasKey(subKey):
+    var elapsedTime = epochTime() - wf.failedPeers[subKey]

```

Comment:
I think it should be possible to use https://github.com/status-im/nim-chronos/blob/master/chronos/timer.nim functions herr, rather than using `std/times` functions e.g. `epochTime()`. These are generally faster and less prone to errors.

For example, you could measure the `elapsedTime` as a `Duration` from `Momen.now()`

Reply:
solved in 2f0e923b3e6c2494bdd639ee6ad352f56bd2f434